### PR TITLE
[v7r1] Fix handling of job states when from requests

### PR DESCRIPTION
**Pull request by phicharp**
BEGINRELEASENOTES

*WMS
FIX: when the job status was set from a failover request and the job was `Stalled`, the job status was by mistake updated to `Running`but this was not visible from the `loggingInfo` that was correct... When finalizing the request, the status of the job was thus left `Running` while the job was either `Done` or `Failed`.

To fix this, we don't update the job status when the request is executed but in the request finalisation we must recover the last but one job status to know what to do...

ENDRELEASENOTES

### CODE REVIEW

**Comment by phicharp** [comment]
I will include the modifications that had been done in v7r1 already... Apologies...

File: RequestManagementSystem/Client/ReqClient.py [hunk 1]
```diff
@@ -17,22 +16,20 @@
...
-from DIRAC.Core.Utilities.JEncode import strToIntDict
-from DIRAC.Core.Utilities.DEncode import ignoreEncodeWarning
 from DIRAC.ConfigurationSystem.Client import PathFinder
-from DIRAC.Core.Base.Client import Client, createClient
+from DIRAC.Core.Base.Client import Client
 from DIRAC.RequestManagementSystem.Client.Request import Request
 from DIRAC.RequestManagementSystem.private.RequestValidator import RequestValidator
 
 
-@createClient('RequestManagement/ReqManager')
```
Comment (fstagni) [inline RequestManagementSystem/Client/ReqClient.py]: Why?

File: RequestManagementSystem/Client/ReqClient.py [hunk 2]
```diff
@@ -240,7 +234,7 @@ def getRequestStatus(self, requestID):
     :param self: self reference
     :param int requestID: id of the request
     """
-    if isinstance(requestID, six.string_types):
+    if isinstance(requestID, basestring):
```
Comment (fstagni) [inline RequestManagementSystem/Client/ReqClient.py]: Mmm @phicharp I think this is the result of a bad rebase...?

File: RequestManagementSystem/Client/ReqClient.py [hunk 3]
```diff
@@ -365,19 +367,12 @@ def getRequestIDsForJobs(self, jobIDs):
                               "Failed" : { jobIDn: errMsg, jobIDm: errMsg, ...}  )
     """
     self.log.verbose("getRequestIDsForJobs: attempt to get request(s) for job %s" % jobIDs)
-    res = self._getRPC().getRequestIDsForJobs(jobIDs)
-    if not res["OK"]:
+    requests = self._getRPC().getRequestIDsForJobs(jobIDs)
+    if not requests["OK"]:
```
Comment (fstagni) [inline RequestManagementSystem/Client/ReqClient.py]: I prefer what was before.

File: RequestManagementSystem/Client/ReqClient.py [hunk 4]
```diff
@@ -449,17 +443,17 @@ def prettyPrint(mainItem, key='', offset=0):
     for item in mainItem:
       prettyPrint(item, offset=offset + 2)
     output += "%s%s\n" % (blanks, ']' if isinstance(mainItem, list) else ')')
-  elif isinstance(mainItem, six.string_types):
+  elif isinstance(mainItem, basestring):
```
Comment (fstagni) [inline RequestManagementSystem/Client/ReqClient.py]: here too.

File: RequestManagementSystem/Client/ReqClient.py (hunk 1 above)
Comment (phicharp) [reply RequestManagementSystem/Client/ReqClient.py]: Yes, sorry I had not rebased correctly...

**Pull request closed without merging**